test/Oscoin/Test/Storage/Block/Equivalence.hs

```diff
@@ -102,28 +101,33 @@ type StoresUnderTest c tx s m =
 -- used internally for the SQLite store implementation (cfr. 'withStores').
 type StoreM c = StateT (Orphanage c DummyTx DummySeal) IO
 
+
 -- | Initialises both the SQL and the STM store and pass them to the 'action'.
 -- For the SQL store, we need to cheat: due to the fact we /do not/ want to
 -- run the 'Protocol' chain selection on both stores (or it would defy the
 -- purpose of these tests) we need to override the 'insertBlock' for the
 -- SQLite store to pass through chain selection.
 withStores
+    :: forall c a.
+       IsCrypto c
+    => (StoresUnderTest c DummyTx DummySeal IO -> IO a)
+    -> IO a
+withStores action =
+    SQLite.withBlockStore ":memory:" defaultGenesis $ \sqlStore ->
+        STM.withBlockStore (fromGenesis defaultGenesis) blockScore $ \stmStore ->
+            action (sqlStore, stmStore)
```

Comment:
It's a shame these two functions are ever-so-slightly different. I think trying to refactor this is going to be slightly tricky due to the callback-style nature of this function, and maybe is not worth it 🤔